Cabal/doc/developing-packages.rst

```diff
@@ -1,83 +1,119 @@
 Quickstart
 ==========
 
-Lets assume we have created a project directory and already have a
-Haskell module or two.
+Starting from scratch, we're going to walk you through creating a simple
+Haskell project.
 
-Every project needs a name, we'll call this example "proglet".
+**TL;DR;** ``mkdir proglet && cd proglet && cabal init --non-interactive && cabal run``
 
-.. highlight:: console
 
-::
+Introduction
+------------
 
-    $ cd proglet/
-    $ ls
-    Proglet.hs
+Every project needs a name, we'll call ours "proglet" and start by
+creating an empty directory.
 
-It is assumed that (apart from external dependencies) all the files that
-make up a package live under a common project root directory. This
-simple example has all the project files in one directory, but most
-packages will use one or more subdirectories.
+.. highlight:: console
 
-To turn this into a Cabal package we need two extra files in the
-project's root directory:
+::
 
--  ``proglet.cabal``: containing package metadata and build information.
+    $ mkdir proglet
+    $ cd proglet/
 
--  ``Setup.hs``: usually containing a few standardized lines of code,
-   but can be customized if necessary.
 
-We can create both files manually or we can use ``cabal init`` to create
-them for us.
+Using ``cabal init``
+--------------------
 
-Using "cabal init"
-------------------
+The ``cabal init`` command creates the necessary files for a Cabal package,
+it has both an ``--interactive`` (default) and ``--non-interactive``
```

Comment:
haven't these reversed?

Reply:
Since I wrote this, yes. Move the (default) to after non-interactive.